open-sauced/app
# fix: a11y fail   click events have key events and no static element interactions

## Description
 Addresses the accessibility problems in the files listed in the linked issue.

## Related Tickets & Documents
Fixes #4130 

## Mobile & Desktop Screenshots/Recordings

**AuthSection:** 
![authsection - after](https://github.com/user-attachments/assets/cc6f3c2e-9e54-408d-9e3a-2d14e564c052)
![authsection -before](https://github.com/user-attachments/assets/1029b9e1-d401-4a6a-aac6-fed63555b04e)

**CardRepoList:**
![cardrepolist - after](https://github.com/user-attachments/assets/fce80439-e048-493d-89e9-913d84b0093b)
![cardrepolist - before](https://github.com/user-attachments/assets/62f41135-c692-403f-9be8-9465093266c9)

**ContributorFilterDropdown:**
![contributorFilterDropdown - after](https://github.com/user-attachments/assets/24661fa2-2c1b-49e5-99f8-126bd26247b4)
![contributorFilterDropdwon - before](https://github.com/user-attachments/assets/2b655088-a619-4cbc-aeb9-d8b1f06348c4)

**ContributorHighLightCard:**
Neutral and hiver states remain same - 
![contributorHighlightCard](https://github.com/user-attachments/assets/e469f11f-bba4-46bf-8d2c-d0f81c7891ff)

![contributorHighloghtCard - hover](https://github.com/user-attachments/assets/c2f1d5fc-5e92-4026-8666-34f402181924)

**ContributorlistTableRow:** 
No diff, except it now has a cursor on hover: 
![contributorlistTableRow - no diff except cursor](https://github.com/user-attachments/assets/0f3a1a70-2276-46c7-b8ea-bb2304e1eeee)

**DevCard:**
![devcard - no diff](https://github.com/user-attachments/assets/e333462c-a071-4499-bce0-5562f93eb207)

**HighlightFilterCard:**
![highlight filtercard - no diff](https://github.com/user-attachments/assets/62105b3d-ecb6-4dc4-b9a1-1c49b24e0db2)

**HighlightInputForm close button:**
![highlightinputform - close button](https://github.com/user-attachments/assets/48bd4dd4-e3cc-41db-94e3-07787357119d)

**HighlightInputForm - suggested links:**
![highlightinputform - suggested highlight link](https://github.com/user-attachments/assets/5a937b07-2603-44d7-948c-ca7781c84ba7)

**PillSelector:**
![pill selector](https://github.com/user-attachments/assets/f0b11dae-dd67-4d07-b6d6-d4dd904ceb4e)

![pill selector after](https://github.com/user-attachments/assets/a167e060-2e83-4565-8b35-64548f40cc59)

**RepositoryCartItem:**
![repositorycartitem - no diff](https://github.com/user-attachments/assets/7c2e9596-664e-42fe-a7b9-12baf6108500)

**Search:**
![search](https://github.com/user-attachments/assets/5b7e6eb9-7fdb-49ff-b399-96611ee13e6a)
![search - focussed](https://github.com/user-attachments/assets/4b892aa5-79d5-4cdf-95cc-7fff4fb898c2)

**Search-dialog:**
![search-dialog - no diff](https://github.com/user-attachments/assets/90576887-c3ae-4b20-83f6-0bba9ce6ba75)

**SuperlativeSelector:** 
Unselected - 
![superlative selector - no diff](https://github.com/user-attachments/assets/15a6db52-3730-451d-97dc-e1c2a07cae80)
Selected - 
![superlative selctor - selectec - no diff](https://github.com/user-attachments/assets/72b95d29-3908-41a6-af8b-a3495978c4f1)

**TextInput:**
![textinput - before](https://github.com/user-attachments/assets/db53a64d-8de4-4a61-a054-8b27a5edc8ec)

![textinput- after](https://github.com/user-attachments/assets/ba45d8c6-987b-47b4-93e4-0d4851e3fc16)

**ToggleOption:**
![toggleoption - after](https://github.com/user-attachments/assets/8c82149a-db63-413d-af59-6f06bf7b0b41)
![toggleoption - before](https://github.com/user-attachments/assets/1209cb44-f500-4f80-aed3-75fb8dcb8748)

With Icon
![toggleoption with icon - after](https://github.com/user-attachments/assets/56fcee8a-2d45-435b-9a4d-48ca8d33c867)
![toggleoption with icon - before](https://github.com/user-attachments/assets/e3963698-bef2-4793-9fed-6c1b10ccb58c)

**Card in  pages/u/[username]/Card.tsx:**
![u:username card - no diff](https://github.com/user-attachments/assets/343a7abf-8e36-4d65-9f96-6bfadcbd4566)

Radio and Radio Check: 
![Screenshot 2024-10-23 at 01 02 01](https://github.com/user-attachments/assets/91272ce8-f8e7-4f9c-a0b2-844d60f5b336)

![Screenshot 2024-10-23 at 01 03 37](https://github.com/user-attachments/assets/ea08bfa9-304c-4169-b2d9-3fa6c2415550)

## Steps to QA
Run `npx eslint .` in the root folder, and see that those errors are no longer occurring


## Tier (staff will fill in)


## [optional] What gif best describes this PR or how it makes you feel?

![A picture of a black woman chewing with her eyes darting nervously](https://media2.giphy.com/media/v1.Y2lkPTc5MGI3NjExamx2M3kzZ2I1bW5pbmdoNWdxZG8xdWo4ODl5aXV2bWY4NzlhejZ2dSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/KQS3zjDesBGpO/giphy.webp)
